.github/workflows/test-native-gradle-plugin.yml

```diff
@@ -22,9 +22,30 @@ concurrency:
   cancel-in-progress: ${{ github.event_name == 'pull_request' || github.repository != 'graalvm/native-build-tools' }}
 
 jobs:
+  populate-matrix:
+    name: "Set matrix"
+    runs-on: "ubuntu-20.04"
+    timeout-minutes: 5
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: "☁️ Checkout repository"
+        uses: actions/checkout@v3
+      - name: "🔧 Prepare environment"
+        uses: graalvm/setup-graalvm@v1
+        with:
+          java-version: '17'
+          distribution: 'graalvm'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: "🕸️ Populate matrix"
+        id: set-matrix
+        run: |
+          ./gradlew -PmatrixType=gradle :native-gradle-plugin:dumpFunctionalTestList
+
   test-native-gradle-plugin:
     name: "Sanity checks"
-    runs-on: ${{ matrix.os }}
+    runs-on: "ubuntu-20.04"
+    timeout-minutes: 60
```

Comment:
where does this timeout come from? One native test shouldn't take longer than maybe 15min, no? Can we use a timeout of 20 or 30min maybe?

Reply:
Now that we have almost final overview how much time jobs take, we can decrease timeout. I checked jobs and the longest one took 30min, so probably 40min should be okay limit. What do you think?